cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -288,22 +300,79 @@ friendlyVotingProcedures
-    redeemerInfo <- friendlyRedeemer shEra tb
-    return ["redeemers" .= redeemerInfo]
-
-friendlyRedeemer :: MonadWarning m => ShelleyBasedEra era -> TxBody era -> m Aeson.Value
-friendlyRedeemer _ (ShelleyTxBody _ _ _ TxBodyNoScriptData _ _) = return Aeson.Null
-friendlyRedeemer _ (ShelleyTxBody _ _ _ (TxBodyScriptData _ _ r) _ _) = encodingToJSON $ L.toCBOR r
+getRedeemerDetails
+  :: forall era m. MonadWarning m => AlonzoEraOnwards era -> TxBody era -> m [Aeson.Pair]
+getRedeemerDetails aeo tb = do
+  let ShelleyTx _ ledgerTx = makeSignedTransaction [] tb
+  redeemerInfo <- friendlyRedeemers ledgerTx
+  return ["redeemers" .= redeemerInfo]
  where
-  encodingToJSON :: MonadWarning m => Encoding -> m Aeson.Value
-  encodingToJSON e =
-    eitherToWarning Aeson.Null $
-      first ("Error decoding redeemer: " ++) $
-        fromFlatTerm (decodeValue True) $
-          toFlatTerm e
+  friendlyRedeemers
+    :: Ledger.Tx (ShelleyLedgerEra era)
+    -> m (Aeson.Value)
+  friendlyRedeemers tx =
+    alonzoEraOnwardsConstraints aeo $ do
+      let redeemerAndExUnitPairList = Map.toList $ Ledger.unRedeemers $ tx ^. Ledger.witsTxL . Ledger.rdmrsTxWitsL
+      redeemerList <- mapM (uncurry $ friendlyRedeemerInfo tx) redeemerAndExUnitPairList
+      return $ Aeson.Array $ Vector.fromList redeemerList
+
+  friendlyRedeemerInfo
+    :: Ledger.Tx (ShelleyLedgerEra era)
+    -> Ledger.PlutusPurpose Ledger.AsIx (ShelleyLedgerEra era)
+    -> (Ledger.Data (ShelleyLedgerEra era), ExUnits)
+    -> m Aeson.Value
+  friendlyRedeemerInfo tx redeemerPurpose (redeemerData, exUnits) =
+    alonzoEraOnwardsConstraints aeo $ do
+      let inputNotFoundError = "Could not find corresponding input to " <> show redeemerPurpose
+          mCorrespondingInput = strictMaybeToMaybe $ Ledger.redeemerPointerInverse (tx ^. Ledger.bodyTxL) redeemerPurpose
+          mFriendlyPurposeResult = friendlyPurpose aeo <$> mCorrespondingInput
+      friendlyPurposeResult <-
+        eitherToWarning (Aeson.Null) $ maybeToEither inputNotFoundError mFriendlyPurposeResult
+      return $
+        object
+          [ "purpose" .= friendlyPurposeResult
+          , "redeemer" .= friendlyRedeemer redeemerData exUnits
+          ]
+
+  friendlyRedeemer :: Ledger.Data (ShelleyLedgerEra era) -> ExUnits -> Aeson.Value
+  friendlyRedeemer scriptData ExUnits{exUnitsSteps = exSteps, exUnitsMem = exMemUnits} =
+    object
+      [ "data" .= Aeson.String (T.pack $ show $ unData scriptData)
+      , "execution units"
+          .= object
+            [ "steps" .= Aeson.Number (fromIntegral exSteps)
+            , "memory" .= Aeson.Number (fromIntegral exMemUnits)
+            ]
+      ]
+
+  friendlyPurpose
+    :: AlonzoEraOnwards era -> Ledger.PlutusPurpose AsIxItem (ShelleyLedgerEra era) -> Aeson.Value
+  friendlyPurpose AlonzoEraOnwardsAlonzo purpose =
+    case purpose of
+      Ledger.AlonzoSpending (Ledger.AsIxItem _ sp) -> Aeson.object ["spending script witnessed input" .= friendlyInput sp]
```

Comment:
You can abstract out the rending of the `Ledger.AsIxItem`s

Reply:
Addressed in: https://github.com/IntersectMBO/cardano-cli/commit/27b2910b10b5069bb4066e130af593ca3a26b3e2